src/lib/steps/package.ts

```diff
@@ -28,105 +28,213 @@ const resolvePaths = (workingDirectory: string, packageConfig: NgPackageConfig):
 
   const defaultPath = path.join(baseDirectory, 'ng-package.json');
-  if (defaultPath !== file) {
+  if (defaultPath !== filePath) {
     // read default ng-package config file
-    promiseChain = promiseChain.then((ngPkg: NgPackageConfig) => {
-
-      return readNgPackageFile(defaultPath)
-        .then((otherNgPkg: NgPackageConfig) => {
-          // merge both ng-package config objects
-          return merge(ngPkg, otherNgPkg, arrayMergeLogic);
-        });
-
+    promiseChain = promiseChain.then(async (ngPkg: NgPackageConfig) => {
+      const otherNgPkg: NgPackageConfig = await readNgPackageFile(defaultPath);
+      // merge both ng-package config objects
+      return merge(ngPkg, otherNgPkg, arrayMergeLogic);
     });
   }
 
-  return promiseChain
-    .then((ngPkg: NgPackageConfig) => {
-      // resolve paths relative to `ng-package.json` file
-      const dir = path.resolve(baseDirectory, ngPkg.src || '.');
+  const ngPkg: NgPackageConfig = await promiseChain;
+  // resolve paths relative to `ng-package.json` file
+  const packageConfigurationDirectory = path.resolve(baseDirectory, ngPkg.src || '.');
+
+  // read 'package.json'
+  log.debug('loading package.json');
+
+  const pkg: any = await readJson(path.resolve(packageConfigurationDirectory, 'package.json'));
+  // merge package.json ng-package config
+  const finalPackageConfig = merge(ngPkg, pkg.ngPackage, arrayMergeLogic);
+  // make sure we provide default values for src and dest
+  finalPackageConfig.src = finalPackageConfig.src || packageConfigurationDirectory;
+  finalPackageConfig.dest = finalPackageConfig.dest || path.resolve(packageConfigurationDirectory,'dist');
+  return new NgPackageData(
+    finalPackageConfig.src,
+    pkg.name,
+    finalPackageConfig.dest,
+    finalPackageConfig.src,
+    finalPackageConfig
+  );
+}
 
-      // read 'package.json'
-      log.debug('loading package.json');
-      return readJson(path.resolve(dir, 'package.json'))
-        .then((pkg: any) => {
-          // merge package.json ng-package config
-          const finalPackageConfig = merge(ngPkg, pkg.ngPackage, arrayMergeLogic);
+async function readSecondaryPackage(rootPackage: NgPackageData, filePath: string): Promise<NgPackageData> {
+  const baseDirectory = path.dirname(filePath);
+  const ngPackageFile = path.resolve(baseDirectory, 'ng-package.json');
+  const packageJsonFile = path.resolve(baseDirectory, 'package.json');
 
-          // read 'ng-package.schema.json'
-          return schemaPromise.then((SchemaClass) => {
-            const schema = new SchemaClass(ngPkg);
+  let ngPackage: NgPackageConfig = await readNgPackageFile(ngPackageFile);
+  const packageJson = await readJson(packageJsonFile);
 
-            return new NgPackage(pkg, finalPackageConfig, dir, schema);
-          });
+  ngPackage = merge(ngPackage, packageJson.ngPackage, arrayMergeLogic);
+  ngPackage.lib.externals = rootPackage.libExternals;
```

Comment:
Searching for ng-package config at `...\packages\components\lib\alert\ng-package.json`

```
Cannot set property 'externals' of undefined
TypeError: Cannot set property 'externals' of undefined
    at ...\node_modules\ng-packagr\lib\steps\package.js:166:33
```

Reply:
Fixed with a simple `if` statement.